x/ibc-rate-limit/ibc_middleware_test.go

```diff
@@ -0,0 +1,350 @@
+	return osmosisApp, app.NewDefaultGenesisState()
+}
+
+func NewTransferPath(chainA, chainB *osmosisibctesting.TestChain) *ibctesting.Path {
+	path := ibctesting.NewPath(chainA.TestChain, chainB.TestChain)
+	path.EndpointA.ChannelConfig.PortID = ibctesting.TransferPort
+	path.EndpointB.ChannelConfig.PortID = ibctesting.TransferPort
+	path.EndpointA.ChannelConfig.Version = transfertypes.Version
+	path.EndpointB.ChannelConfig.Version = transfertypes.Version
+	return path
+}
+
+func (suite *MiddlewareTestSuite) SetupTest() {
+	suite.Setup()
+	ibctesting.DefaultTestingAppInit = SetupTestingApp
+	suite.coordinator = ibctesting.NewCoordinator(suite.T(), 2)
+	suite.chainA = &osmosisibctesting.TestChain{
+		TestChain: suite.coordinator.GetChain(ibctesting.GetChainID(1)),
+	}
+	// Remove epochs to prevent  minting
+	suite.chainA.MoveEpochsToTheFuture()
+	suite.chainB = &osmosisibctesting.TestChain{
+		TestChain: suite.coordinator.GetChain(ibctesting.GetChainID(2)),
+	}
+	suite.path = NewTransferPath(suite.chainA, suite.chainB)
+	suite.coordinator.Setup(suite.path)
+}
+
+// Helpers
+
+// NewValidMessage generates a new sdk.Msg of type MsgTransfer.
+// forward=true means that the message will be a "send" message, while forward=false is for  a "receive" message.
+// amount represents the amount transferred
+func (suite *MiddlewareTestSuite) NewValidMessage(forward bool, amount sdk.Int) sdk.Msg {
+	var coins sdk.Coin
+	var port, channel, accountFrom, accountTo string
+
+	if forward {
+		coins = sdk.NewCoin(sdk.DefaultBondDenom, amount)
+		port = suite.path.EndpointA.ChannelConfig.PortID
+		channel = suite.path.EndpointA.ChannelID
+		accountFrom = suite.chainA.SenderAccount.GetAddress().String()
+		accountTo = suite.chainB.SenderAccount.GetAddress().String()
+	} else {
+		coins = transfertypes.GetTransferCoin(
+			suite.path.EndpointB.ChannelConfig.PortID,
+			suite.path.EndpointB.ChannelID,
+			sdk.DefaultBondDenom,
+			sdk.NewInt(1),
+		)
+		coins = sdk.NewCoin(sdk.DefaultBondDenom, amount)
+		port = suite.path.EndpointB.ChannelConfig.PortID
+		channel = suite.path.EndpointB.ChannelID
+		accountFrom = suite.chainB.SenderAccount.GetAddress().String()
+		accountTo = suite.chainA.SenderAccount.GetAddress().String()
+	}
+
+	timeoutHeight := clienttypes.NewHeight(0, 100)
+	return transfertypes.NewMsgTransfer(
+		port,
+		channel,
+		coins,
+		accountFrom,
+		accountTo,
+		timeoutHeight,
+		0,
+	)
+}
+
+// Tests that a receiver address longer than 4096 is not accepted
+func (suite *MiddlewareTestSuite) TestInvalidReceiver() {
+	msg := transfertypes.NewMsgTransfer(
+		suite.path.EndpointB.ChannelConfig.PortID,
+		suite.path.EndpointB.ChannelID,
+		sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(1)),
+		suite.chainB.SenderAccount.GetAddress().String(),
+		strings.Repeat("x", 4097),
+		clienttypes.NewHeight(0, 100),
+		0,
+	)
+	ack, _ := suite.ExecuteReceive(msg)
+	suite.Require().Contains(string(ack), "error",
+		"acknoledgment is not an error")
```

Comment:
```suggestion
		"acknowledgment is not an error")
```

Reply:
ahh, I can never write acknowledgement right!  I'll update that in the error too so that the test doesn't fail